visualization/motion_planning_tasks/src/task_list_model.cpp

```diff
@@ -251,7 +251,8 @@ void TaskListModel::processTaskDescriptionMessage(const std::string& id,
 	// retrieve existing or insert new remote task for given id
 	auto it_inserted = remote_tasks_.insert(std::make_pair(id, nullptr));
 	bool created = it_inserted.second;
-	RemoteTaskModel*& remote_task = it_inserted.first->second;
+	auto& task_it = it_inserted.first;
```

Comment:
nitpick: I think, we don't need a reference here. An iterator is essentially a pointer anyway and we don't want to risk to overwrite the iterator accidentally.

Reply:
By value would be (slighly) more efficient, because we don't need an additional dereference...

Reply:
No compiler would care with reasonable optimizations switched on.
By semantics, it's a shorter name for a more complex expression, not a copied value.

Reply:
I don't think so. A reference is essentially a pointer, which requires a dereference to access its contents. This cannot be optimized away by the compiler. By the benefit will we marginal in any case.

Reply:
Spot the difference: https://godbolt.org/z/C58u38
Point of fact, there is none.